File: leetcode/python/0367_valid_perfect_square.py
class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        l = 0
        r = num
        while l <= r:
            m = (l + r) >> 1
            val = m * m

            if val == num: return True
            elif val > num: r = m - 1
            else: l = m + 1

        return False

File: leetcode/cpp/0367_valid_perfect_square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        if (num == 1) {
            return true;
        }
        
        for (int i = 1; i <= num / 2; i++) {
            if (long(long(i) * long(i)) == long(num)) {
                return true;
            }
        }
        
        return false;
    }
};